### PR TITLE
Implemented ElementFactory

### LINKED ISSUE #15: Implement ElementFactory

### PR DESCRIPTION
**Pull request by mialeska**
##### Related Issue Link: 

Closes #15 

##### Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/main/java/aquality/selenium/core/elements/ElementFactory.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+                case MORE_THEN_ZERO:
+                    conditionalWait.waitForTrue(() -> !elementFinder.findElements(locator, state, ZERO_TIMEOUT).isEmpty(),
+                            localizationManager.getLocalizedMessage("loc.no.elements.found.by.locator",
+                                    locator.toString(), state.toString()));
+                    break;
+                case Any:
+                    conditionalWait.waitFor(() -> elementFinder.findElements(locator, state, ZERO_TIMEOUT) != null);
+                    break;
+                default:
+                    throw new IllegalArgumentException("No such expected value:".concat(count.toString()));
```
Comment (DmitryBogatko) [inline src/main/java/aquality/selenium/core/elements/ElementFactory.java]: add space after ':"' in the message

File: src/main/java/aquality/selenium/core/elements/ElementsCount.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package aquality.selenium.core.elements;
+
+public enum ElementsCount {
+    ZERO,
+    MORE_THEN_ZERO,
+    Any
```
Comment (DmitryBogatko) [inline src/main/java/aquality/selenium/core/elements/ElementsCount.java]: Any -> ANY to have the same style

File: src/main/java/aquality/selenium/core/elements/interfaces/IElement.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+public interface IElement extends IParent {
+
+    /**
+     * Gets clear WebElement.
+     *
+     * @return WebElement
+     */
+    default RemoteWebElement getElement() {
+        return getElement(null);
```
Comment (DmitryBogatko) [inline src/main/java/aquality/selenium/core/elements/interfaces/IElement.java]: what does initializing with null mean?

Comment (mialeska) [reply src/main/java/aquality/selenium/core/elements/interfaces/IElement.java]: it means that the timeout wasn't specified and we will use the default one

File: src/main/java/aquality/selenium/core/elements/ElementsCount.java (hunk 2 above)
Comment (mialeska) [reply src/main/java/aquality/selenium/core/elements/ElementsCount.java]: fixed

File: src/main/java/aquality/selenium/core/elements/ElementFactory.java (hunk 1 above)
Comment (mialeska) [reply src/main/java/aquality/selenium/core/elements/ElementFactory.java]: fixed

**Pull request merged**